### PR TITLE
Updated `oid` and `column_order` fields to only allow positive integers

### LINKED ISSUE #3176: Update Constraints for Positive Integers in `oid` and `column_order`
Currently, there is an issue with the constraints defined for the `oid` field in the Constraint object and the `column_order` field in the TableSettings object. Both of these fields have constraints that allow negative integers. 
For the `Objectid`(oid) field in the Constraint object, it is important to ensure that the identifier remains a positive integer. Similarly, the `column_order` field in the TableSettings object, which specifies the order of columns, should only accept positive integers.

### PR DESCRIPTION
**Pull request by varshadr**
Fixes #3176 

This PR addresses constraints for the `oid` field  and the `column_order` field in the TableSettings object. Both fields currently allow negative integers. This update ensures that `oid`  and `column_order` only accepts positive integers.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by silentninja — CHANGES_REQUESTED** [review]
@varshadr Please run [`makemigrations`](https://docs.djangoproject.com/en/4.2/ref/django-admin/#django-admin-makemigrations) to generate the migration file for converting the existing database columns and add it to the PR.

**Comment by varshadr** [comment]
> @varshadr Please run [`makemigrations`](https://docs.djangoproject.com/en/4.2/ref/django-admin/#django-admin-makemigrations) to generate the migration file for converting the existing database columns and add it to the PR.

Okay 👍

**Comment by seancolsen** [comment]
@varshadr I'm re-assigning this to @mathemancer because Mukesh will no longer be working with you on this project. Please assign any further PRs to @mathemancer for review.

**Comment by varshadr** [comment]
> @varshadr I'm re-assigning this to @mathemancer because Mukesh will no longer be working with you on this project. Please assign any further PRs to @mathemancer for review.

Sure 👍

**Comment by mathemancer** [comment]
After looking more closely, this PR isn't actually documentation, though it's related to the overall project. Reopening, I think we should try to review and merge this bugfix.

**Review by mathemancer — CHANGES_REQUESTED** [review]
Hey, @varshadr, this looks good to me, except there's two migrations. To fix that, please:

- Clear out your docker state
- Pull this branch
- Merge develop into it
- Remove all migrations added by this PR
- Rerun the makemigrations command.
- Commit the resulting mitration file.

**Comment by mathemancer** [comment]
@varshadr Also, please make sure the tests are passing.

**Comment by varshadr** [comment]
> @varshadr Also, please make sure the tests are passing.

Sure, I'll check

**Comment by Anish9901** [comment]
Hey @varshadr, are you facing any problems with this issue? Do you need help?

**Comment by varshadr** [comment]
> Hey @varshadr, are you facing any problems with this issue? Do you need help?

Hey @Anish9901. Yes, I need help with the failing tests. Could you help me identify the cause for the failing tests?

File: mathesar/models/base.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
 from mathesar.state import make_sure_initial_reflection_happened, get_cached_metadata, reset_reflection
 from mathesar.state.cached_property import cached_property
 from mathesar.api.exceptions.database_exceptions.base_exceptions import ProgrammingAPIException
+from django.contrib.postgres.fields import ArrayField
```
Comment (Anish9901) [inline mathesar/models/base.py:55]: We've decided to support `sqlite` for our internal database going forward. The `ArrayField` is a postgres specific field and isn't compatible with sqlite. Therefore, you should stick with `JSONField` for `column_order`.

File: mathesar/models/base.py [hunk 2]
```diff
@@ -901,7 +902,7 @@ class PreviewColumnSettings(BaseModel):
 class TableSettings(ReflectionManagerMixin, BaseModel):
     preview_settings = models.OneToOneField(PreviewColumnSettings, on_delete=models.CASCADE)
     table = models.OneToOneField(Table, on_delete=models.CASCADE, related_name="settings")
-    column_order = JSONField(null=True, default=None)
+    column_order = column_order = ArrayField(models.PositiveIntegerField(), null=True, default=None)
```
Comment (Anish9901) [inline mathesar/models/base.py:905]: If you are facing issues while using `PositiveIntegerField()` with `JSONField`, you could look into writing your own validator, see [this](https://docs.djangoproject.com/en/5.0/ref/validators/) for reference.

**Review by Anish9901 — CHANGES_REQUESTED** [review]
@varshadr please look through the requested changes below, and reach out if you are facing any issues.

**Comment by varshadr** [comment]
Thanks @Anish9901! I'll look into it

File: mathesar/models/base.py (hunk 2 above)
Comment (varshadr) [reply mathesar/models/base.py]: > If you are facing issues while using `PositiveIntegerField()` with `JSONField`, you could look into writing your own validator, see [this](https://docs.djangoproject.com/en/5.0/ref/validators/) for reference.

I implemented the `validate_column_order` function and added it as a validator for the `column_order` field. This ensures all elements in it are positive integers, as discussed. Could you review it @Anish9901 ?

**Comment by Anish9901** [comment]
This is looking great @varshadr! Can you add some tests to verify that the validator is working as intended?

**Comment by seancolsen** [comment]
I put this in the v0.1.4 milestone because we'd like to #3296 and this PR introduces migrations.

**Comment by Anish9901** [comment]
@varshadr for the sake of merging this PR I am taking over writing the test cases, please let me know if you have any concerns.

**Comment by varshadr** [comment]
> @varshadr for the sake of merging this PR I am taking over writing the test cases, please let me know if you have any concerns.

Sure, you could go on with it

**Review by Anish9901 — APPROVED** [review]

**Pull request merged**